WPMedia/fusion-news-theme-blocks
# ASUB-7841 Subscriptions Paywall - migrating from v1 to v2

**If you have not filled out the checklist below, the pr is not ready for review.**

## Description

Migrating Subscriptions paywall block from v1 to v2

## Jira Ticket

https://arcpublishing.atlassian.net/browse/ASUB-7841

## Acceptance Criteria

_copy from ticket_

## Test Steps

Before starting you need additional configuration on `Composer` side

1. Create a Paywall status list
[https://[env].[org].arcpublishing.com/composer/picklist/content_restrictions](https://[env].[org].arcpublishing.com/composer/picklist/content_restrictions)

<img width="1578" alt="Screenshot 2023-11-17 at 8 44 02" src="https://github.com/WPMedia/arc-themes-blocks/assets/54566275/e0e0b383-3045-4823-a1b4-efa1b877a8ad">

Each article to be evaluated (according to the paywall rules) should have at least 3 attributes as part of their ANS: `type`, `canonical_url`, `content_restrictions.content_code`

2. In Subscriptions, our paywall is running on the client side. You will need to deploy a paywall with the rules you want to be applied. You are able to create paywall rules by going into Subscriptions app
Go into the Paywall Tab
a. Create Rule(s)
b. Create ruleset 
c. Deploy a paywall (Status: active)

<img width="1577" alt="Screenshot 2023-11-17 at 8 40 05" src="https://github.com/WPMedia/arc-themes-blocks/assets/54566275/48ebe87d-7f3f-46d7-9ca3-c9d2d88aebe2">

Each time the paywall is deployed, we are pushing a p.js file to s3. This file contains information about the rules added to the paywall, and the way in which the rules will be evaluated on the client side. 

3. Create the resolver
4. Create the output-type >> Here, you will be able to download the p.js hosted on S3.
https://${org}-${arcSite}-${envKey}.web.arc-cdn.net/arc/subs/p.js

```
Import React from ‘react
import { useAppContext, useFusionContext } from "fusion:context";
import { ENVIRONMENT } from "fusion:environment";

const PwsOutputType = (props) => {
 const scriptSrc = useScriptSrc()
 
 return (
   <html>
     <head>
       <title>{props.metaValue('title')}</title>
       <props.MetaTags />
       <props.Libs />
       <script src={scriptSrc} async id="p-tag"></script>
       <props.CssLinks />
       <link rel='icon' type='image/x-icon' href={props.deployment(`${props.contextPath}/resources/img/favicon.ico`)} />
     </head>
     <body>
       <div id='fusion-app' >
         {props.children}
       </div>
       <props.Fusion />
     </body>
   </html>
 )
}

const useScriptSrc = () => {
 const { contextPath } = useFusionContext()
 const { arcSite } = useAppContext();
 const isLocal = ENVIRONMENT === "localhost"
 const envKey = ["localhost", "subssinglesite-sandbox"].includes(ENVIRONMENT) ? "sandbox" : "prod";
 const scriptSrc = isLocal ? `${contextPath}/resources/p.js` : `https://subssinglesite-${arcSite}-${envKey}.web.arc-cdn.net/arc/subs/p.js`

 return scriptSrc
}

export default PwsOutputType
```
5. Create a template and include the `paywall block` as part of footer
<img width="1578" alt="Screenshot 2023-11-17 at 8 57 18" src="https://github.com/WPMedia/arc-themes-blocks/assets/54566275/6d16c3f7-1b30-47c9-9fd5-a046de58362a">


6. You will need to create several stories, and according to the rules added to the paywall. You should be able to see the regwall or paywall.

7. On subscriptions side, You need an offer published. 

<img width="1575" alt="Screenshot 2023-11-17 at 11 51 37" src="https://github.com/WPMedia/arc-themes-blocks/assets/54566275/e53f8d44-51e2-4c9f-a75b-18e37a2eee96">


## Effect Of Changes

### Before

### After

<img width="1433" alt="Screenshot 2023-11-17 at 9 01 06" src="https://github.com/WPMedia/arc-themes-blocks/assets/54566275/ef8a79ef-cc56-42ae-bfa8-d47f43749897">

<img width="1435" alt="Screenshot 2023-11-17 at 9 01 30" src="https://github.com/WPMedia/arc-themes-blocks/assets/54566275/8bd12354-3620-4235-b0a8-ffd283e99a98">



## Dependencies or Side Effects

_Examples of dependencies or side effects are:_

- Manifest pr for new block:
- Feature pack pr for local development:
- How to deploy to news: https://arcpublishing.atlassian.net/wiki/spaces/TI/pages/3138682964/News+Theme+2.0+Migration+Development+Deployment+Notes
- Additional settings that will be required in the blocks.json
- Changes to the custom fields which will require users to reconfigure features
- Dependency on another PR that needs to be merged first

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

        please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._
